mmdet3d/models/dense_heads/shape_aware_head.py

```diff
@@ -0,0 +1,640 @@
+import numpy as np
+import torch
+from mmcv.cnn import ConvModule, bias_init_with_prob, normal_init
+from torch import nn as nn
+
+from mmdet3d.core import box3d_multiclass_nms, limit_period, xywhr2xyxyr
+from mmdet.core import multi_apply
+from mmdet.models import HEADS
+from ..builder import build_head
+from .anchor3d_head import Anchor3DHead
+
+
+@HEADS.register_module()
+class BaseShapeHead(nn.Module):
+    """Base Shape-aware Head in Shape Signature Network.
+
+    Args:
+        num_cls (int): Number of classes.
+        num_base_anchors (int): Number of anchors per location.
+        box_code_size (int): The dimension of boxes to be encoded.
+        in_channels (int): Input channels for convolutional layers.
+        out_channels (int): Output channels for shared convolutional layers.
+            Default: 64.
+        use_direction_classifier (bool): Whether to use direction classifier.
+            Default: True.
+        conv_cfg (dict): Config of conv layer.
+            Default: dict(type='Conv2d')
+        norm_cfg (dict): Config of norm layer.
+            Default: dict(type='naiveSyncBN2d', eps=1e-3, momentum=0.01).
+        bias (bool|str): Type of bias. Default: False.
+    """
+
+    def __init__(self,
+                 num_cls,
+                 num_base_anchors,
```

Comment:
We may add an `num_shared_convs` to decide the number of shared convs, thus unifying small and large head.

Reply:
Simplified and refactored.